consumption.tex

```diff
@@ -0,0 +1,89 @@
+
+\import{./}{algorithms/alg.verify-tx.tex}
+
+Before looking at how the verification works for the whole proof $\beta$ we focus on the part of the proof pertaining to the transaction, specifically $\alpha = \sf(tx, txInclusion, blk)$. Verifying this part of the proof for some event $e$ is straightforward by making use of the blockchain transaction protocol in Definition~\ref{def:blockchain-tx}. We ensure that the transaction \textsf{tx} indeed contains the value transfer claimed in the provided event $e$, and that the transaction is really contained in the block provided. The function, \textsf{verify-tx}, can be seen in detail in Algorithm~\ref{alg.verify-tx}.
+
+\begin{definition}[Merkle Tree protocol]
+  A \emph{merkle tree} protocol $\mathcal{MT}$ consists of two functions $\mathcal{MT}_t(x_1, \dots, x_t)$ and $\VerMT_t(root, x, \pi)$ which work as follows:
+
+  \begin{itemize}
+    \item $\mathcal{MT}_t(x_1, \dots, x_t)$: Given elements $x_1, \dots, x_t$, return the root of the Merkle Tree with exactly $t$ leaves, where for each $1 \le i \le t$, the $i$-th leaf corresponds to $x_i$.
+
+    \item $\VerMT(root, x, \pi)$: Given a Merkle Tree root, an element $x$ and an inclusion proof $\pi$, return $\sf{true}$ if and only if $\pi$ is a valid proof for the fact that $x$ is contained in the tree rooted at $root$.
+  \end{itemize}
+\end{definition}
+
+\import{./}{algorithms/alg.verify-event-federation.tex}
+
+Now that the issue of verifying the transaction is covered, we move our attention to the issue of verifying that the provided block \textsf{blk} really belongs to the best source blockchain. Let us assume that the smart contract holds in its state the continuously updated set of all valid blocks on the best source blockchain, in the form of a Merkle Tree root $\mathcal{M}$. Then, by verifying a Merkle Tree inclusion proof \textsf{blockConnection} for $\textsf{blk} \in \mathcal{M}$, such that $\VerMT(\mathcal{M}, \textsf{blk}, \textsf{blockConnection}) = \true$, a verifier can be certain that the block provided is indeed part of the best source blockchain. Combined with verifying the transaction this completes our $\textsf{verify}_\mathcal{M}^e$ function, shown in Algorithm~\ref{alg.verify-event-federation}.
```

Comment:
I'm confused about when we switched from a decentralized setting (in which a contestation period is required) to a federation setting...